### PR TITLE
RelevateHealth Bid Adapter : Initial release

### PR DESCRIPTION
**Pull request by hasanideepak**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

**Comment by jsnellbaker** [comment]
Docs PR:
https://github.com/prebid/prebid.github.io/pull/5377

File: test/spec/modules/relevatehealthBidAdapter_spec.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      expect(data[0].us_privacy).to.equal('1NYN');
+      //   let _bidRequest = {};
+      //   let _Request1 = spec.buildRequests(request, _bidRequest);
+      //   let data1 = JSON.parse(_Request1.data);
+      //   expect(data1.regs).to.equal(undefined);
+    });
+  });
+  describe('Validate response ', function () {
+    it('Validate bid response : valid bid response', function () {
+      let bRequest = spec.buildRequests(request);
```
Comment (spotxslagle) [inline test/spec/modules/relevatehealthBidAdapter_spec.js:154]: Correct me if I'm wrong, but it doesn't seem like setting bRequest and data is actually doing anything in this test. Remove these two lines.

**Review by spotxslagle — CHANGES_REQUESTED** [review]
I just had a question I don't think will require changes and then a couple of suggestions for the spec. Otherwise it looks good

File: modules/relevatehealthBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  // Loop through each bid request
+  bidRequests.forEach(bid => {
+    // Construct the bid request object
+    const request = {
+      id: generateUUID(),
+      placementId: bid.params.placementId,
+      imp: [{
+        id: bid.bidId,
+        banner: getBanner(bid),
+        bidfloor: bid.params.bid_floor
```
Comment (patmmccann) [inline modules/relevatehealthBidAdapter.js:20]: Please also look to the getFloor function in the request

File: modules/relevatehealthBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    } catch (err) {
+      logError(err);
+    }
+  }
+  return resp;
+}
+
+// Function to check if Bid is valid
+function isBidRequestValid(bid) {
+  return !!(bid.params.placementId && bid.params.user_id);
```
[the remark this answers is not in the document]
Comment (hasanideepak) [reply modules/relevatehealthBidAdapter.js]: To make all consistent, changed placementId to placement_id. Also changed in Docs

File: test/spec/modules/relevatehealthBidAdapter_spec.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+          bidder: 'relevatehealth',
+          params: {
+            placementId: 110011,
+            user_id: '11211'
+          }
+        },
+        isValid = spec.isBidRequestValid(bid);
+      expect(isValid).to.equals(true);
+    });
+    it('isBidValid : placementId and user_id are not passed', function () {
```
[the remark this answers is not in the document]
Comment (hasanideepak) [reply test/spec/modules/relevatehealthBidAdapter_spec.js]: Added the validations as asked.

File: test/spec/modules/relevatehealthBidAdapter_spec.js (hunk 1 above)
Comment (hasanideepak) [reply test/spec/modules/relevatehealthBidAdapter_spec.js]: Yes, you are right. Removed the lines.

File: modules/relevatehealthBidAdapter.js (hunk 2 above)
Comment (hasanideepak) [reply modules/relevatehealthBidAdapter.js]: Created a function getFloor() to get bid_floor.

**Comment by ChrisHuie** [comment]
@hasanideepak can you please fix indentation issues with your pr. Linting is not liking it right now. Thx

**Review by spotxslagle — APPROVED** [review]
Once the linting is fixed LGTM

**Comment by hasanideepak** [comment]
> @hasanideepak can you please fix indentation issues with your pr. Linting is not liking it right now. Thx

Made changes for indentation issues.

**Comment by hasanideepak** [comment]
> Once the linting is fixed LGTM

linting issue is fixed.

**Comment by ChrisHuie** [comment]
@hasanideepak indentation and new lines seem to still be an issue here

**Comment by hasanideepak** [comment]
> @hasanideepak indentation and new lines seem to still be an issue here

Hi, tried to solve the issue. Please review it and if issue is still there then it would be helpful if you can please specify the file and  show some lines with issue examples.

**Comment by spotxslagle** [comment]
If you click on the details of the ci/circleci: build section you can see all the logs.

/home/circleci/Prebid.js/test/spec/modules/relevatehealthBidAdapter_spec.js
    2:1  error  Expected indentation of 2 spaces but found 4    indent
    5:1  error  Expected indentation of 2 spaces but found 4    indent
   10:1  error  Expected indentation of 2 spaces but found 4    indent
   11:1  error  Expected indentation of 2 spaces but found 4    indent
   13:1  error  Expected indentation of 2 spaces but found 4    indent
   14:1  error  Expected indentation of 4 spaces but found 8    indent
   15:1  error  Expected indentation of 6 spaces but found 12   indent
   16:1  error  Expected indentation of 6 spaces but found 12   indent
   17:1  error  Expected indentation of 8 spaces but found 16   indent
   18:1  error  Expected indentation of 10 spaces but found 20  indent
   19:1  error  Expected indentation of 12 spaces but found 24  indent
   20:1  error  Expected indentation of 10 spaces but found 20  indent
   21:1  error  Expected indentation of 8 spaces but found 16   indent
   22:1  error  Expected indentation of 6 spaces but found 12   indent
   23:1  error  Expected indentation of 6 spaces but found 12   indent
   24:1  error  Expected indentation of 8 spaces but found 16   indent
   25:1  error  Expected indentation of 8 spaces but found 16   indent
   26:1  error  Expected indentation of 8 spaces but found 16   indent
   27:1  error  Expected indentation of 8 spaces but found 16   indent
   28:1  error  Expected indentation of 8 spaces but found 16   indent
   29:1  error  Expected indentation of 8 spaces but found 16   indent
   30:1  error  Expected indentation of 6 spaces but found 12   indent
   31:1  error  Expected indentation of 4 spaces but found 8    indent
   32:1  error  Expected indentation of 4 spaces but found 8    indent
   33:1  error  Expected indentation of 6 spaces but found 12   indent
   34:1  error  Expected indentation of 8 spaces but found 16   indent
   35:1  error  Expected indentation of 8 spaces but found 16   indent
   36:1  error  Expected indentation of 10 spaces but found 20  indent
   37:1  error  Expected indentation of 12 spaces but found 24  indent
   38:1  error  Expected indentation of 12 spaces but found 24  indent
   39:1  error  Expected indentation of 12 spaces but found 24  indent
   40:1  error  Expected indentation of 12 spaces but found 24  indent
   41:1  error  Expected indentation of 12 spaces but found 24  indent
   42:1  error  Expected indentation of 12 spaces but found 24  indent
   43:1  error  Expected indentation of 12 spaces but found 24  indent
   44:1  error  Expected indentation of 12 spaces but found 24  indent
   45:1  error  Expected indentation of 12 spaces but found 24  indent
   46:1  error  Expected indentation of 12 spaces but found 24  indent
   47:1  error  Expected indentation of 12 spaces but found 24  indent
   48:1  error  Expected indentation of 12 spaces but found 24  indent
   49:1  error  Expected indentation of 10 spaces but found 20  indent
   50:1  error  Expected indentation of 10 spaces but found 20  indent
   51:1  error  Expected indentation of 10 spaces but found 20  indent
   52:1  error  Expected indentation of 8 spaces but found 16   indent
   53:1  error  Expected indentation of 8 spaces but found 16   indent
   54:1  error  Expected indentation of 8 spaces but found 16   indent
   55:1  error  Expected indentation of 6 spaces but found 12   indent
   56:1  error  Expected indentation of 4 spaces but found 8    indent
   57:1  error  Expected indentation of 4 spaces but found 8    indent
   58:1  error  Expected indentation of 6 spaces but found 12   indent
   59:1  error  Expected indentation of 8 spaces but found 16   indent
   60:1  error  Expected indentation of 8 spaces but found 16   indent
   61:1  error  Expected indentation of 10 spaces but found 20  indent
   62:1  error  Expected indentation of 12 spaces but found 24  indent
   63:1  error  Expected indentation of 12 spaces but found 24  indent
   64:1  error  Expected indentation of 12 spaces but found 24  indent
   65:1  error  Expected indentation of 12 spaces but found 24  indent
   66:1  error  Expected indentation of 12 spaces but found 24  indent
   67:1  error  Expected indentation of 12 spaces but found 24  indent
   68:1  error  Expected indentation of 12 spaces but found 24  indent
   69:1  error  Expected indentation of 12 spaces but found 24  indent
   70:1  error  Expected indentation of 12 spaces but found 24  indent
   71:1  error  Expected indentation of 12 spaces but found 24  indent
   72:1  error  Expected indentation of 12 spaces but found 24  indent
   73:1  error  Expected indentation of 12 spaces but found 24  indent
   74:1  error  Expected indentation of 10 spaces but found 20  indent
   75:1  error  Expected indentation of 10 spaces but found 20  indent
   76:1  error  Expected indentation of 10 spaces but found 20  indent
   77:1  error  Expected indentation of 8 spaces but found 16   indent
   78:1  error  Expected indentation of 8 spaces but found 16   indent
   79:1  error  Expected indentation of 8 spaces but found 16   indent
   80:1  error  Expected indentation of 6 spaces but found 12   indent
   81:1  error  Expected indentation of 4 spaces but found 8    indent
   82:1  error  Expected indentation of 2 spaces but found 4    indent
   84:1  error  Expected indentation of 2 spaces but found 4    indent
   85:1  error  Expected indentation of 4 spaces but found 8    indent
   86:1  error  Expected indentation of 6 spaces but found 12   indent
   87:1  error  Expected indentation of 10 spaces but found 20  indent
   88:1  error  Expected indentation of 10 spaces but found 20  indent
   89:1  error  Expected indentation of 12 spaces but found 24  indent
   90:1  error  Expected indentation of 12 spaces but found 24  indent
   91:1  error  Expected indentation of 10 spaces but found 20  indent
   92:1  error  Expected indentation of 8 spaces but found 16   indent
   93:1  error  Expected indentation of 8 spaces but found 16   indent
   94:1  error  Expected indentation of 6 spaces but found 12   indent
   95:1  error  Expected indentation of 4 spaces but found 8    indent
   96:1  error  Expected indentation of 4 spaces but found 8    indent
   97:1  error  Expected indentation of 6 spaces but found 12   indent
   98:1  error  Expected indentation of 10 spaces but found 20  indent
   99:1  error  Expected indentation of 10 spaces but found 20  indent
  100:1  error  Expected indentation of 12 spaces but found 24  indent
  101:1  error  Expected indentation of 12 spaces but found 24  indent
  102:1  error  Expected indentation of 12 spaces but found 24  indent
  103:1  error  Expected indentation of 12 spaces but found 24  indent
  104:1  error  Expected indentation of 10 spaces but found 20  indent
  105:1  error  Expected indentation of 8 spaces but found 16   indent
  106:1  error  Expected indentation of 8 spaces but found 16   indent
  107:1  error  Expected indentation of 6 spaces but found 12   indent
  108:1  error  Expected indentation of 4 spaces but found 8    indent
  109:1  error  Expected indentation of 4 spaces but found 8    indent
  110:1  error  Expected indentation of 6 spaces but found 12   indent
  111:1  error  Expected indentation of 10 spaces but found 20  indent
  112:1  error  Expected indentation of 10 spaces but found 20  indent
  113:1  error  Expected indentation of 12 spaces but found 24  indent
  114:1  error  Expected indentation of 12 spaces but found 24  indent
  115:1  error  Expected indentation of 12 spaces but found 24  indent
  116:1  error  Expected indentation of 12 spaces but found 24  indent
  117:1  error  Expected indentation of 12 spaces but found 24  indent
  118:1  error  Expected indentation of 10 spaces but found 20  indent
  119:1  error  Expected indentation of 8 spaces but found 16   indent
  120:1  error  Expected indentation of 8 spaces but found 16   indent
  121:1  error  Expected indentation of 6 spaces but found 12   indent
  122:1  error  Expected indentation of 4 spaces but found 8    indent
  123:1  error  Expected indentation of 4 spaces but found 8    indent
  124:1  error  Expected indentation of 6 spaces but found 12   indent
  125:1  error  Expected indentation of 10 spaces but found 20  indent
  126:1  error  Expected indentation of 10 spaces but found 20  indent
  127:1  error  Expected indentation of 12 spaces but found 24  indent
  128:1  error  Expected indentation of 12 spaces but found 24  indent
  129:1  error  Expected indentation of 12 spaces but found 24  indent
  130:1  error  Expected indentation of 12 spaces but found 24  indent
  131:1  error  Expected indentation of 12 spaces but found 24  indent
  132:1  error  Expected indentation of 10 spaces but found 20  indent
  133:1  error  Expected indentation of 8 spaces but found 16   indent
  134:1  error  Expected indentation of 8 spaces but found 16   indent
  135:1  error  Expected indentation of 6 spaces but found 12   indent
  136:1  error  Expected indentation of 4 spaces but found 8    indent
  137:1  error  Expected indentation of 2 spaces but found 4    indent
  138:1  error  Expected indentation of 2 spaces but found 4    indent
  139:1  error  Expected indentation of 4 spaces but found 8    indent
  140:1  error  Expected indentation of 6 spaces but found 12   indent
  141:1  error  Expected indentation of 8 spaces but found 16   indent
  142:1  error  Expected indentation of 6 spaces but found 12   indent
  143:1  error  Expected indentation of 4 spaces but found 8    indent
  144:1  error  Expected indentation of 4 spaces but found 8    indent
  145:1  error  Expected indentation of 6 spaces but found 12   indent
  146:1  error  Expected indentation of 6 spaces but found 12   indent
  147:1  error  Expected indentation of 6 spaces but found 12   indent
  148:1  error  Expected indentation of 6 spaces but found 12   indent
  149:1  error  Expected indentation of 4 spaces but found 8    indent
  150:1  error  Expected indentation of 4 spaces but found 8    indent
  151:1  error  Expected indentation of 6 spaces but found 12   indent
  152:1  error  Expected indentation of 6 spaces but found 12   indent
  153:1  error  Expected indentation of 6 spaces but found 12   indent
  154:1  error  Expected indentation of 6 spaces but found 12   indent
  155:1  error  Expected indentation of 4 spaces but found 8    indent
  156:1  error  Expected indentation of 4 spaces but found 8    indent
  157:1  error  Expected indentation of 6 spaces but found 12   indent
  158:1  error  Expected indentation of 6 spaces but found 12   indent
  159:1  error  Expected indentation of 6 spaces but found 12   indent
  160:1  error  Expected indentation of 6 spaces but found 12   indent
  161:1  error  Expected indentation of 6 spaces but found 12   indent
  162:1  error  Expected indentation of 4 spaces but found 8    indent
  163:1  error  Expected indentation of 4 spaces but found 8    indent
  164:1  error  Expected indentation of 6 spaces but found 12   indent
  165:1  error  Expected indentation of 6 spaces but found 12   indent
  166:1  error  Expected indentation of 6 spaces but found 12   indent
  167:1  error  Expected indentation of 6 spaces but found 12   indent
  168:1  error  Expected indentation of 4 spaces but found 8    indent
  169:1  error  Expected indentation of 4 spaces but found 8    indent
  170:1  error  Expected indentation of 6 spaces but found 12   indent
  171:1  error  Expected indentation of 8 spaces but found 16   indent
  172:1  error  Expected indentation of 6 spaces but found 12   indent
  173:1  error  Expected indentation of 6 spaces but found 12   indent
  174:1  error  Expected indentation of 6 spaces but found 12   indent
  175:1  error  Expected indentation of 6 spaces but found 12   indent
  176:1  error  Expected indentation of 4 spaces but found 8    indent
  177:1  error  Expected indentation of 2 spaces but found 4    indent
  178:1  error  Expected indentation of 2 spaces but found 4    indent
  179:1  error  Expected indentation of 4 spaces but found 8    indent
  180:1  error  Expected indentation of 6 spaces but found 12   indent
  181:1  error  Expected indentation of 6 spaces but found 12   indent
  182:1  error  Expected indentation of 6 spaces but found 12   indent
  183:1  error  Expected indentation of 6 spaces but found 12   indent
  184:1  error  Expected indentation of 6 spaces but found 12   indent
  185:1  error  Expected indentation of 6 spaces but found 12   indent
  186:1  error  Expected indentation of 6 spaces but found 12   indent
  187:1  error  Expected indentation of 6 spaces but found 12   indent
  188:1  error  Expected indentation of 6 spaces but found 12   indent
  189:1  error  Expected indentation of 6 spaces but found 12   indent
  190:1  error  Expected indentation of 6 spaces but found 12   indent
  191:1  error  Expected indentation of 6 spaces but found 12   indent
  192:1  error  Expected indentation of 4 spaces but found 8    indent
  193:1  error  Expected indentation of 4 spaces but found 8    indent
  194:1  error  Expected indentation of 6 spaces but found 12   indent
  195:1  error  Expected indentation of 6 spaces but found 12   indent
  196:1  error  Expected indentation of 6 spaces but found 12   indent
  197:1  error  Expected indentation of 4 spaces but found 8    indent
  198:1  error  Expected indentation of 2 spaces but found 4    indent
  199:1  error  Expected indentation of 2 spaces but found 4    indent
  200:1  error  Expected indentation of 4 spaces but found 8    indent
  201:1  error  Expected indentation of 6 spaces but found 12   indent
  202:1  error  Expected indentation of 8 spaces but found 16   indent
  203:1  error  Expected indentation of 10 spaces but found 20  indent
  204:1  error  Expected indentation of 10 spaces but found 20  indent
  205:1  error  Expected indentation of 8 spaces but found 16   indent
  206:1  error  Expected indentation of 6 spaces but found 12   indent
  207:1  error  Expected indentation of 6 spaces but found 12   indent
  208:1  error  Expected indentation of 6 spaces but found 12   indent
  209:1  error  Expected indentation of 6 spaces but found 12   indent
  210:1  error  Expected indentation of 6 spaces but found 12   indent
  211:1  error  Expected indentation of 4 spaces but found 8    indent
  212:1  error  Expected indentation of 4 spaces but found 8    indent
  213:1  error  Expected indentation of 6 spaces but found 12   indent
  214:1  error  Expected indentation of 8 spaces but found 16   indent
  215:1  error  Expected indentation of 10 spaces but found 20  indent
  216:1  error  Expected indentation of 12 spaces but found 24  indent
  217:1  error  Expected indentation of 12 spaces but found 24  indent
  218:1  error  Expected indentation of 10 spaces but found 20  indent
  219:1  error  Expected indentation of 8 spaces but found 16   indent
  220:1  error  Expected indentation of 6 spaces but found 12   indent
  221:1  error  Expected indentation of 6 spaces but found 12   indent
  222:1  error  Expected indentation of 6 spaces but found 12   indent
  223:1  error  Expected indentation of 6 spaces but found 12   indent
  224:1  error  Expected indentation of 6 spaces but found 12   indent
  225:1  error  Expected indentation of 4 spaces but found 8    indent
  226:1  error  Expected indentation of 4 spaces but found 8    indent
  227:1  error  Expected indentation of 6 spaces but found 12   indent
  228:1  error  Expected indentation of 8 spaces but found 16   indent
  229:1  error  Expected indentation of 10 spaces but found 20  indent
  230:1  error  Expected indentation of 12 spaces but found 24  indent
  231:1  error  Expected indentation of 10 spaces but found 20  indent
  232:1  error  Expected indentation of 8 spaces but found 16   indent
  233:1  error  Expected indentation of 6 spaces but found 12   indent
  234:1  error  Expected indentation of 6 spaces but found 12   indent
  235:1  error  Expected indentation of 6 spaces but found 12   indent
  236:1  error  Expected indentation of 6 spaces but found 12   indent
  237:1  error  Expected indentation of 4 spaces but found 8    indent
  238:1  error  Expected indentation of 2 spaces but found 4    indent
  239:4  error  Newline required at end of file but not found   eol-last

✖ 778 problems (354 errors, 424 warnings)
  354 errors and 2 warnings potentially fixable with the `--fix` option.

**Comment by hasanideepak** [comment]
> If you click on the details of the ci/circleci: build section you can see all the logs.

@spotxslagle Thank you for the details. I modifed the code as suggested please review it. Thanks

**Comment by hasanideepak** [comment]
@spotxslagle @ChrisHuie : indentation and new line issue is resolved now and errors in ci/circleci are not related to this adapter. Please let me know next step.

**Pull request merged**